packages/components/src/modal/index.tsx

```diff
@@ -172,10 +209,19 @@ function UnforwardedModal(
 					<div
 						className={ classnames( 'components-modal__content', {
 							'hide-header': __experimentalHideHeader,
+							'is-scrollable': hasScrollableContent,
 							'has-scrolled-content': hasScrolledContent,
 						} ) }
 						role="document"
 						onScroll={ onContentContainerScroll }
+						ref={ contentRef }
+						aria-label={
+							hasScrollableContent
+								? scrollableContentLabel ||
+								  __( 'Scrollable section' )
+								: undefined
+						}
+						tabIndex={ hasScrollableContent ? 0 : undefined }
```

Comment:
Do you think we can extract all this logic into a hook `const { ref, props } = useScrollable();` 

and just basically apply the "ref" and "props" to the element that is supposed to be scrollable? I think it might bring some clarity to the code here.

Reply:
Also, not sure why we need `childrenContainerRef` can't we just check the height of the scrollable itself (or at least its children) ?

Reply:
The scrollable element has a fixed height,
The height of the content (the children) determines the actual content overflow, when it's greater than the scrollable height.
There may be multipole children though. Not sure how to calculate eh height of multiple children that could also be dynamically rendered. Seems to me the simplest option is to check for the height of an additional container.

Reply:
> Do you think we can extract all this logic into a hook 

I'm all for better abstraction and improvements. I'd prefer to do that when enhancing the `Scrollable` component though. See https://github.com/WordPress/gutenberg/issues/45809